ext/cl_khr_extended_versioning.asciidoc

```diff
@@ -0,0 +1,284 @@
+for applications to process strings to test for the presence of an extension or
+intermediate language or built-in kernel.
+
+=== General information
+
+==== Name Strings
+
+`cl_khr_extended_versioning`
+
+==== Contributors
+
+Kévin Petit, Arm Ltd. +
+Ben Ashbaugh, Intel +
+Alastair Murray, Codeplay Software Ltd. +
+Einar Hov, Arm Ltd.
+
+==== Version history
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2020-02-12 | 1.0.0     | Initial version.
+|====
+
+==== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 2.2, Revision 11.
+
+This extension requires OpenCL 1.0.
+
+=== New API Types
+
+==== Version
+
+This extension introduces a new scheme to encode detailed
+(major, minor, patch/revision) version information into a single 32-bit unsigned
+integer:
+
+* The major version is using bits 31-22
+* The minor version is using bits 21-12
+* The patch version is using bits 11-0
+
+This scheme enables two versions to be ordered using the standard C/C++ operators.
+Macros are provided to extract individual fields or compose a full version
+from the individual fields.
+
+[source,c]
+----
+
+typedef cl_uint cl_version_khr;
+
+#define CL_VERSION_MAJOR_BITS_KHR (10)
+#define CL_VERSION_MINOR_BITS_KHR (10)
+#define CL_VERSION_PATCH_BITS_KHR (12)
+
+#define CL_VERSION_MAJOR_MASK_KHR ((1 << CL_VERSION_MAJOR_BITS_KHR) - 1)
+#define CL_VERSION_MINOR_MASK_KHR ((1 << CL_VERSION_MINOR_BITS_KHR) - 1)
+#define CL_VERSION_PATCH_MASK_KHR ((1 << CL_VERSION_PATCH_BITS_KHR) - 1)
+
+#define CL_VERSION_MAJOR_KHR(version) \
+        ((version) >> (CL_VERSION_MINOR_BITS_KHR + CL_VERSION_PATCH_BITS_KHR))
+#define CL_VERSION_MINOR_KHR(version) \
+        (((version) >> CL_VERSION_PATCH_BITS_KHR) & CL_VERSION_MINOR_MASK_KHR)
+#define CL_VERSION_PATCH_KHR(version) ((version) & CL_VERSION_PATCH_MASK_KHR)
+
+#define CL_MAKE_VERSION_KHR(major, minor, patch) \
+    ((((major) & CL_VERSION_MAJOR_MASK_KHR) << (CL_VERSION_MINOR_BITS_KHR + CL_VERSION_PATCH_BITS_KHR)) | \
+     (((minor) & CL_VERSION_MINOR_MASK_KHR) << CL_VERSION_PATCH_BITS_KHR) | \
+     ((patch) & CL_VERSION_PATCH_MASK_KHR))
+----
+
+==== Name and version
+
+This extension adds a structure that can be used to describe a combination of a
+name alongside a version number:
+
+[source,c]
+----
+#define CL_NAME_VERSION_MAX_NAME_SIZE_KHR 64
+
+typedef struct _cl_name_version_khr {
+    cl_version_khr version;
+    char name[CL_NAME_VERSION_MAX_NAME_SIZE_KHR];
+} cl_name_version_khr;
+----
+
+The `name` field is an array of `CL_NAME_VERSION_MAX_NAME_SIZE_KHR` bytes used as
+storage for a NUL-terminated string whose maximum length is therefore
+`CL_NAME_VERSION_MAX_NAME_SIZE_KHR - 1`.
+
+=== New API Enums
+
+Accepted value for the _param_name_ parameter to *clGetPlatformInfo*:
+
+[source,c]
+----
+CL_PLATFORM_NUMERIC_VERSION_KHR         0x0906
+CL_PLATFORM_EXTENSIONS_WITH_VERSION_KHR 0x0907
```

Comment:
See previous comment regarding including enum values in the extension spec.

Reply:
I'm all for consistency. Are you proposing we don't add them here and do it all later or keep them here and create issues to follow up?

Reply:
Either could work but I have a slight preference for this so the spec sources stay consistent:

> we don't add them here and do it all later

I can't find the commit ID off-hand but I think I removed extension enum assignments from a handful of extensions previously.

Reply:
Ok, I've removed all of them.